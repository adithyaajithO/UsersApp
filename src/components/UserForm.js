import React from 'react';
import { Text, View, TouchableOpacity, StyleSheet, TextInput } from 'react-native';


const UserForm = ({ userName, password, setUser, setPassword, buttonLabel, onSubmit }) => {

    return <>
        <View style={styles.inputContainerStyle}>
            <Text style={styles.textStyle}>User ID: </Text>
            <TextInput
                value={userName}
                onChangeText={(value) => {
                    setUser(value)
                }}
                style={styles.inputStyle}
                autoCapitalize="none"
                autoCorrect={false} />
        </View>
        <View style={styles.inputContainerStyle}>
            <Text style={styles.textStyle}>Password: </Text>
            <TextInput
                value={password}
                onChangeText={(value) => {
                    setPassword(value)
                }}
                style={styles.inputStyle}
                secureTextEntry
                autoCapitalize="none"
                autoCorrect={false} />
        </View>
        <TouchableOpacity style={{ marginVertical: 30 }}
            onPress={() => onSubmit()}>
            <Text style={styles.buttonLabelStyle}>{buttonLabel}</Text>
        </TouchableOpacity>
    </>
}

const styles = StyleSheet.create({
    inputContainerStyle: {
        flexDirection: "row",
        height: 40,
        marginBottom: 3
    },
    inputStyle: {
        flex: 1,
        backgroundColor: "rgb(244,244,244)",
        borderRadius: 4,
        paddingHorizontal: 7
    },
    textStyle: {
        alignSelf: 'center',
        width: 75,
        fontSize: 15
    },
    buttonLabelStyle: {
        fontSize: 18,
        color: '#3798E9',
        alignSelf: 'center'
    }
});

export default UserForm;