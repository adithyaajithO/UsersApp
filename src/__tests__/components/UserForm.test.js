import React from 'react';
import renderer from 'react-test-renderer';

import UserForm from '../../components/UserForm';

test('renders correctly', () => {
    const tree = renderer.create(<UserForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });