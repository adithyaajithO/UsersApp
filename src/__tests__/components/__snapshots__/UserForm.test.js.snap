// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 3,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "fontSize": 15,
          "width": 75,
        }
      }
    >
      User ID: 
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "rgb(244,244,244)",
          "borderRadius": 4,
          "flex": 1,
          "paddingHorizontal": 7,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>,
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 3,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "fontSize": 15,
          "width": 75,
        }
      }
    >
      Password: 
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "rgb(244,244,244)",
          "borderRadius": 4,
          "flex": 1,
          "paddingHorizontal": 7,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>,
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginVertical": 30,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#3798E9",
          "fontSize": 18,
        }
      }
    />
  </View>,
]
`;
